### PR TITLE
feat: unload parquet with version as `created_by`.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

1. by default,  it is `concat!("parquet-rs version ", env!("CARGO_PKG_VERSION"))`
2. since we have our own extention types, and only databend know how to interpreter them, we should mark the file as `databend-created` .
3.  we need add some additional metadata, and it may change.  then we can know how to read files generated by older version.  for example,  we now can read column IS variant type,  but not Has variant  (https://github.com/datafuselabs/databend/issues/13385), obviously, we need a new way to do so.

https://github.com/datafuselabs/databend/issues/13385

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [x] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15680)

### CODE REVIEW

File: src/query/storages/stage/src/append/parquet_file/writer_processor.rs [hunk 1]
```diff
@@ -83,6 +84,7 @@ fn create_writer(
         .set_dictionary_enabled(false)
         .set_statistics_enabled(EnabledStatistics::None)
         .set_bloom_filter_enabled(false)
+        .set_created_by(DATABEND_COMMIT_VERSION.to_string())
```
Comment (b41sh) [inline src/query/storages/stage/src/append/parquet_file/writer_processor.rs:87]: Why do we need using `DATABEND_COMMIT_VERSION` as parquet `created_by`?

Comment (youngsofun) [reply src/query/storages/stage/src/append/parquet_file/writer_processor.rs:87]: Supplemented in the summary

**Comment by youngsofun** [comment]
error just because the unload file size changed.
I will update them after pr is approved.

**Review by b41sh — APPROVED** [review]

**Review by sundy-li — APPROVED** [review]

**Comment by youngsofun** [comment]
I will the version string like `Databend v1.2.261-nightly-a2fcca8d1d(rust-1.78.0-nightly-2024-05-30T08:45:18.525800000Z)` with simply ``Databend v1.2.261-nightly` if no problem wit you @b41sh @sundy-li

**Pull request merged**